src/helm_client.rs

```diff
@@ -0,0 +1,181 @@
+use fluvio_command::CommandExt;
+use std::process::Command;
+use tracing::{instrument, warn};
+
+use super::Chart;
+use super::HelmError;
+use super::InstallArg;
+use super::InstalledChart;
+use super::UninstallArg;
+
+/// Client to manage helm operations
+#[derive(Debug)]
+#[non_exhaustive]
+pub struct HelmClient {}
+
+impl HelmClient {
+    /// Creates a Rust client to manage our helm needs.
+    ///
+    /// This only succeeds if the helm command can be found.
+    pub fn new() -> Result<Self, HelmError> {
+        let output = Command::new("helm").arg("version").result()?;
+
+        // Convert command output into a string
+        let out_str = String::from_utf8(output.stdout).map_err(HelmError::Utf8Error)?;
+
+        // Check that the version command gives a version.
+        // In the future, we can parse the version string and check
+        // for compatible CLI client version.
+        if !out_str.contains("version") {
+            return Err(HelmError::HelmVersionNotFound(out_str));
+        }
+
+        // If checks succeed, create Helm client
+        Ok(Self {})
+    }
+
+    /// Installs the given chart under the given name.
+    ///
+    #[instrument(skip(self))]
+    pub fn install(&self, args: &InstallArg) -> Result<(), HelmError> {
+        let mut command = args.install();
+        command.result()?;
+        Ok(())
+    }
+
+    /// Upgrades the given chart
+    #[instrument(skip(self))]
+    pub fn upgrade(&self, args: &InstallArg) -> Result<(), HelmError> {
+        let mut command = args.upgrade();
+        command.result()?;
+        Ok(())
+    }
+
+    /// Uninstalls specified chart library
+    pub fn uninstall(&self, uninstall: UninstallArg) -> Result<(), HelmError> {
+        if uninstall.ignore_not_found {
+            let app_charts = self
+                .get_installed_chart_by_name(&uninstall.release, uninstall.namespace.as_deref())?;
+            if app_charts.is_empty() {
+                warn!("Chart does not exists, {}", &uninstall.release);
+                return Ok(());
+            }
+        }
+        let mut command: Command = uninstall.into();
+        command.result()?;
+        Ok(())
+    }
+
+    /// Adds a new helm repo with the given chart name and chart location
+    #[instrument(skip(self))]
+    pub fn repo_add(&self, chart: &str, location: &str) -> Result<(), HelmError> {
```

Comment:
Seems like repo related commands should be move to it's own builder type?

Reply:
Tracking feedback in https://github.com/infinyon/fluvio-helm/issues/21

I think it would be good to tackle this in another PR since this one is purely refactoring